### ISSUE TITLE
Emmet suggestion not appearing at the top of the list in jsx files

### ISSUE THREAD

**Issue by rajinder-yadav** [Microsoft/vscode#39518]
- VSCode Version: Code - Insiders 1.19.0-insider (5b0fa2eddbe82df90fb07eea120c9447c93dbd2f, 2017-12-01T06:07:42.474Z)
- OS Version: Darwin x64 17.2.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
vscode-markdownlint|Dav|0.12.0
EditorConfig|Edi|0.11.1
vscode-eslint|dba|1.4.3
tslint|eg2|1.0.24
cpptools|ms-|0.14.3
debugger-for-chrome|msj|3.5.0
autoimport|ste|1.5.3
vim|vsc|0.10.5
markdown-pdf|yza|0.1.7

(2 theme extensions excluded)

When I type form and hit tab, I expect it  to get emmet to expand it to the <form> tags. Instead it is replaced with the word "from".

I have notice this happens with other tags and emmet expansion is unreliable for one off keywords. It would be nice to switch the behaviour for emmet to kick in with hitting tab 2 times really quickly, so intellisense don't do the wrong thing.

---

Steps to Reproduce:

1. in a .jsx file type 'form'
2. hit tab for emmet expansion
3. form is replaced with from

![screen shot 2017-12-03 at 8 53 26 pm 2](https://user-images.githubusercontent.com/439226/33533077-cb035b8e-d86c-11e7-9b7a-f79a19af8b85.png)

Reproduces without extensions: Yes/No

**Comment by rajinder-yadav** [comment]
Even if I hit escape to dismiss intellisense, then hit tab, emmet expansion doesn't kick in.

**Comment by rajinder-yadav** [comment]
Also I should always be able to type a html tag, or user defined react, angular tag, like "<from" hit tab and emmet should know what to do when fail on a keyword!

**Comment by ramya-rao-a** [comment]
Where is this `from` keyword coming from? I can't get a repro. If I can repro this, I can look into why the `from` is sorted higher compared to `form`.

You can also add the below 2 settings to ensure emmet suggestions always appear on top
```
 "emmet.showSuggestionsAsSnippets": true,
    "editor.snippetSuggestions": "top"
```

> Even if I hit escape to dismiss intellisense, then hit tab, emmet expansion doesn't kick in.

Emmet expansion on tab in the absence of intellisense occurs only if you have added the setting `"emmet.triggerExpansionOnTab": true`

**Comment by rajinder-yadav** [comment]
No idea where from is coming from and why?
Try creating a simple react.js app, then create a new component file with a .jsx extension to enable emmet for this file. If you're not sure how to do this, just ask me to create a simple project you can fetch from github.

**Comment by rajinder-yadav** [comment]
Also thanks for the setting suggestions, I added, `"emmet.triggerExpansionOnTab": true` but don't want to use the other for now.

**Comment by rajinder-yadav** [comment]
FYI: I did try all 3 settings you suggested and still from get selected even though it's not the top item!

**Comment by ramya-rao-a** [comment]
Please share a sample project where I can try and get a repro.

Also, try running `code --disable-extensions` and see if you still see the `from`. That will tell us if the suggestion is coming from one of your installed extensions

**Comment by rajinder-yadav** [comment]
Using `--disable-extensions` still show the bug for me. Please make sure you remove those 3 vscode setting to showed earlier when investigating.

Here is the demo app: https://github.com/rajinder-yadav/demo_app

Open file `user-form.jsx`, inside the render function, type, "form", you should see the bug. GL :)

**Comment by ramya-rao-a** [comment]
Thanks for the sample code @rajinder-yadav, I can repro it now.
My mistake, I was trying to repro in stable, whereas this issue only appears in Insiders
The root cause is the we recently implemented a feature request #15419 that allows typos to accommodate typos.

@jrieken Here is a case where relaxing for typos is counter productive. In this case both `form` and `from` get the same score. Shouldnt the non typo match be scored higher than the one for the typo?
